src/mlpack/tests/nmf_test.cpp

```diff
@@ -222,4 +222,66 @@ BOOST_AUTO_TEST_CASE(SparseNMFALSTest)
       1e-5);
 }
 
-BOOST_AUTO_TEST_SUITE_END();
+/**
+ * Check if all elements in W and H are non-negative.
+ * Default Case.
+ * Random Acol initilization and distance minimization update.
+ */
+BOOST_AUTO_TEST_CASE(NonNegNMFDefaultTest)
+{
+  mat w = randu<mat>(20, 12);
+  mat h = randu<mat>(12, 20);
+  mat v = w * h;
+  const size_t r = 12;
+
+  AMF<> nmf;
+  nmf.Apply(v, r, w, h);
+
+  BOOST_REQUIRE(arma::all(vectorise(w) >= 0)
+      && arma::all(vectorise(h) >= 0));
+}
+
+/**
+ * Check if all elements in W and H are non-negative.
+ * Random initialization divergence minimization update.
+ */
+BOOST_AUTO_TEST_CASE(NonNegNMFRandomDivTest)
+{
+  mat w = randu<mat>(20, 12);
+  mat h = randu<mat>(12, 20);
+  mat v = w * h;
+  const size_t r = 12;
+
+  // Custom tighter tolerance.
+  SimpleResidueTermination srt(1e-8, 10000);
+  AMF<SimpleResidueTermination,
+      RandomInitialization,
+      NMFMultiplicativeDivergenceUpdate> nmf(srt);
+  nmf.Apply(v, r, w, h);
+
+  BOOST_REQUIRE(arma::all(vectorise(w) >= 0)
+      && arma::all(vectorise(h) >= 0));
+}
+
+/**
+ * Check if all elements in W and H are non-negative.
+ * Random initialization, alternating least squares update.
+ */
+BOOST_AUTO_TEST_CASE(NonNegNMFALSTest)
+{
+  mat w = randu<mat>(20, 12);
+  mat h = randu<mat>(12, 20);
+  mat v = w * h;
+  const size_t r = 12;
+
+  SimpleResidueTermination srt(1e-12, 50000);
+  AMF<SimpleResidueTermination,
+      RandomAcolInitialization<>,
+      NMFALSUpdate> nmf(srt);
+  nmf.Apply(v, r, w, h);
+
+  BOOST_REQUIRE(arma::all(vectorise(w) >= 0)
+      && arma::all(vectorise(h) >= 0));
```

Comment:
Just a minor note---I don't think that `vectorise()` is needed; if I'm wrong and it is, could you write the more explicit `arma::vectorise()` please?  Some compilers won't accept the plain `vectorise()` since we do not have `using namespace arma;` specified.  Thanks!

Reply:
`vectorise()` is needed because for a matrix `w`, `arma::all(w > 0)` returns a vector, as explained in armadillo document: http://arma.sourceforge.net/docs.html#all
I will change it to `arma::vectorise`.

Reply:
``all(mat)`` returns ``umat``, so in this case I think ``vectorise`` is an easy solution.

Reply:
Yep, thank you for the clarification; I see now that the `vectorise()` is needed.